File: m/rls/stkop.r
file	st_kop - copy counted string
include rid:rider
include rid:stdef

code	st_kop - copy counted

  func	st_kop
	src : * char
	dst : * char
	cnt : int			; buffer length (including zero)
	()  : * char			; at terminator
  is	reply dst if cnt eq		; crazy
	while cnt-- ne			; got more
	  quit if (*dst++ = *src++) eq	;
	end				;
	*--dst = 0			; force termination
	reply dst			;
  end


File: m/rls/rlb/stkop.c
/* file -  st_kop - copy counted string */
#include "m:\rid\rider.h"
#include "m:\rid\stdef.h"
/* code -  st_kop - copy counted */
char *st_kop(
char *src ,
char *dst ,
int cnt )
{ if( cnt == 0)return ( dst );
  while ( cnt-- != 0) {
    if( (*dst++ = *src++) == 0)break;
  } 
  *--dst = 0;
  return ( dst);
} 
